From 9f1c0b0d9a293211e6215cb7ee4d965e534ba3f4 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:24:15 -0400
Subject: [cpufreq] intel_pstate: Remove duplicate CPU ID check

Message-id: <1409235879-1283-162-git-send-email-prarit@redhat.com>
Patchwork-id: 89520
O-Subject: [RHEL7.1 PATCH BZ 1134639 161/185] cpufreq: intel_pstate: Remove duplicate CPU ID check
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 830bcac4e483d347087ed51eed564d1467e1e363
Author: Stratos Karafotis <stratosk@semaphore.gr>
Date:   Tue Jun 10 00:00:09 2014 +0300

    cpufreq: intel_pstate: Remove duplicate CPU ID check

    We check the CPU ID during driver init. There is no need
    to do it again per logical CPU initialization.

    So, remove the duplicate check.

    Signed-off-by: Stratos Karafotis <stratosk@semaphore.gr>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Acked-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index d4d8a26..df554b5 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -690,14 +690,8 @@ MODULE_DEVICE_TABLE(x86cpu, intel_pstate_cpu_ids);
 
 static int intel_pstate_init_cpu(unsigned int cpunum)
 {
-
- const struct x86_cpu_id *id;
  struct cpudata *cpu;
 
- id = x86_match_cpu(intel_pstate_cpu_ids);
- if (!id)
-  return -ENODEV;
-
  all_cpu_data[cpunum] = kzalloc(sizeof(struct cpudata), GFP_KERNEL);
  if (!all_cpu_data[cpunum])
   return -ENOMEM;
-- 
1.7.1